### ISSUE TITLE
PPO regresses the critic onto normalized advantages, so `returns` is not the discounted return

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#97]
`thinkrl/algorithms/ppo.py:227-228` builds the critic's regression target out of advantages that have already been whitened:

```python
advantages = self.compute_gae_advantages(dense_rewards, old_values)
returns = advantages + old_values
```

`BaseRLHFAlgorithm.compute_gae_advantages` normalizes by default, and both `PPOConfig.normalize_advantages` and the base class default are `True`, so `advantages` comes back with mean 0 and unit variance. Adding `old_values` to that does not reconstruct the discounted return, it produces a quantity with no units. `ValueLoss` at `:309` then regresses the critic onto it.

Reproduced on a two-sequence batch with terminal rewards `10.0` and `-4.0` against values `1.0` and `2.0`:

```
shipped returns:  [1.844, 1.910, 1.981, 1.130, 1.089, 1.046]
correct returns:  [8.941, 9.455, 10.000, -3.346, -3.663, -4.000]
```

The terminal-token target is `1.98` where the true return is `10.0`. The critic converges toward the batch statistics of the advantage distribution rather than toward the value function, which means the baseline it supplies on the next iteration is uninformative and the advantages built from it degrade further.

`thinkrl/algorithms/vapo.py:250-253` computes the same quantity correctly:

```python
lambdas_critic = torch.ones(batch_size, device=device) * self.config.lambda_value
adv_critic = self.compute_adaptive_gae(dense_rewards, old_values, lambdas_critic)
returns = adv_critic + old_values
```

VAPO's `compute_adaptive_gae` does not normalize, so its `returns` are the real thing. The two files disagree, which is what suggests this is a slip in `ppo.py` rather than an intentional choice.

**Suggested fix**

Compute the advantages once without normalization, derive `returns` from those, and normalize afterwards for the policy term only. `ppo.py:253-254` already re-normalizes per minibatch, so the normalization inside `compute_gae_advantages` is redundant for the policy path as well.

I am happy to open a PR for this.
